script.py

```diff
@@ -119,98 +170,344 @@ def _load_safety_principles(self) -> List[str]:
+
+        # if we don't display the evaluations we empty the containers
+        else: 
+            for container in containers:
+                container.empty()
+
+
+def toggle_msg_order():
+    st.session_state.reversed = not st.session_state.reversed
 
 def main():
-    st.title("examine|AI")
-    st.subheader("Your assistant for trustworthy conversations with AI")
-    st.image("logo.png", width=200)
+    st.set_page_config(page_title="examine|AI", 
+                       page_icon="🧞‍♂️", 
+                       layout="wide", 
+                       menu_items={
+                           'Get Help': 'https://www.betterhelp.com/',
+                           'Report a bug': "https://wikiroulette.co/",
+                           'About': "### ExamineAI 2023"
+    }
+                       )
+    st.session_state.theme = "dark"
+
+    col1, col2 = st.columns([1, 5])  # Adjust the ratio as needed
+    with col1:
+        st.image("logo.png", width=130)
+        
+    with col2:
+        st.title("examine|AI")
+        st.subheader("Your assistant for trustworthy conversations with AI")
+
+    st.write('<div style="margin-top: 5em;"></div>', unsafe_allow_html=True)
 
+    # Define SafeguardAI object
+    responder = OpenAIResponder(api_key=openai.api_key)
+    safeguard_ai = SafeguardAI(responder)
+    
     # Initialize session state
     MessageStorage()  # Ensures messages are initialized in session state
-    if "last_ai_response" not in st.session_state:
+    if 'last_ai_response' not in st.session_state:
         st.session_state.last_ai_response = None
-    if "evaluate_safeguard" not in st.session_state:
-        st.session_state.evaluate_safeguard = False
-
-    col1, col2 = st.columns(2)
+    if 'evaluate_pressed' not in st.session_state:
+        st.session_state.evaluate_pressed = False
+    if 'disable_eval_button' not in st.session_state:
+        st.session_state.disable_eval_button = False
+    if 'obtain_evaluation' not in st.session_state:
+        st.session_state.obtain_evaluation = False
+    if 'display_evaluation' not in st.session_state:
+        st.session_state.display_evaluation = False
+    if 'reversed' not in st.session_state:
+        st.session_state.reversed = False
+    if 'expandable' not in st.session_state:
+        st.session_state.expandable = False
+    if 'evals' not in st.session_state:
+        st.session_state.evals = " "
+    if 'user_input_to_be_processed' not in st.session_state:
+        st.session_state.user_input_to_be_processed = False
+    if 'show_user_input' not in st.session_state:
+        st.session_state.show_user_input = True
+    if 'disable_user_input' not in st.session_state:
+        st.session_state.disable_user_input = False
+    if 'waiting_for_AI_response' not in st.session_state:
+        st.session_state.waiting_for_AI_response = False
+    if 'AI_response_received' not in st.session_state:
+        st.session_state.AI_response_received = False
```

Comment:
I suggest defining a default state dict that can be easily expanded which we loop through so that we do not have to add/remove more if statements if we modify any of the code logic:

```suggestion
      # Initial state to be given if not found in session state
      default_session_state = {
          'evaluate_pressed': False,
          'disable_eval_button': False,
          'obtain_evaluation': False,
          'display_evaluation': False,
          'reversed': False,
          'expandable': False,
          'evals': " ",
          'user_input_to_be_processed': False,
          'show_user_input': True,
          'disable_user_input': False,
          'waiting_for_AI_response': False,
          'AI_response_received': False
      }
      
      
      for key, default_value in default_session_state.items():
          if key not in st.session_state:
              st.session_state[key] = default_value
```
Streamlit docu says session state values can also be set like attributes so this should work but once again, I was too lazy to actually run this :D